app/models/job_proxy_dispatcher.rb

```diff
@@ -144,11 +175,13 @@ def assign_proxy_to_job(proxy, job)
     end
   end
 
-  def pending_jobs
+  def pending_jobs(target_class = VmOrTemplate)
```

Comment:
@jrafanie Can I rest assured that all the existing users of this method need VmOrTemplate?
(I just don'y know this code well enough to be sure)